mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -429,4 +656,25 @@ void mlirtrt::runtime::registerExecutorTensorRTModuleLuaRuntimeMethods(
                                          *context, stream, va);
         SET_LUA_ERROR_IF_ERROR(result, state);
       };
+
+  lua["_trtrt_enqueue_alloc"] =
+      [allocTracker, resourceTracker](
+          sol::this_state state,
+          std::shared_ptr<NvInferExecContextWrapper> context,
+          CudaStreamPtr stream, uintptr_t outputDesc, sol::table va) {
+        ADD_TENSORRT_MODULE_RANGE("trtrt_enqueue_alloc");
+        sol::state_view luaState(state);
+        assert(context != nullptr);
+        assert(stream != nullptr && "expected valid stream");
+
+        OutputDescriptor desc(outputDesc);
+        auto &allocator = luaState["_trtrt_create_allocator"]
```

Comment:
This whole indirection isn't necessary since this allocator doesn't need to outlive this function (I guess this is just leftover from when the output allocator thing was tied to the pointer lifetime)?

So we can just do `auto allocator = std::make_unique<OutputAllocator>(nbResults);` here?